firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -1077,17 +1077,18 @@ public void resumingAQueryShouldUseBloomFilterToAvoidFullRequery() throws Except
       }
       assertWithMessage("createdDocuments").that(createdDocuments).hasSize(100);
 
-      // Delete 50 of the 100 documents. Use a WriteBatch, rather than DocumentReference.delete(),
-      // to avoid affecting the local cache.
+      // Delete 50 of the 100 documents. Use a different Firestore instance to avoid affecting the
+      // local cache.
       HashSet<String> deletedDocumentIds = new HashSet<>();
       {
-        WriteBatch writeBatchForDocumentDeletes = collection.getFirestore().batch();
+        FirebaseFirestore db2 = testFirestore();
```

Comment:
Does this testFirestore() do the same? getting different instances to same firestore?

Reply:
Yes! If curious about the details, see https://github.com/firebase/firebase-android-sdk/blob/d91342e4b919a89084f5369c47d047bbcc85b385/firebase-firestore/src/androidTest/java/com/google/firebase/firestore/testutil/IntegrationTestUtil.java.